clang/lib/Sema/SemaTemplateDeductionGuide.cpp

```diff
@@ -390,7 +393,7 @@ struct ConvertConstructorToDeductionGuideTransform {
               /*EvaluateConstraint=*/false);
         }
 
-        assert(NewParam->getTemplateDepth() == 0 &&
+        assert(getDepthAndIndex(NewParam).first == 0 &&
```

Comment:
A heads-up: we're encountering clang crashes on this assertion. I'm working on a minimal reproducer.

Reply:
> A heads-up: we're encountering clang crashes on this assertion. I'm working on a minimal reproducer.

Filed https://github.com/llvm/llvm-project/issues/128691.